crates/daphne-server/tests/e2e/test_runner.rs

```diff
@@ -327,11 +330,16 @@ impl TestRunner {
         }
     }
 
-    pub fn report_interval(interval: &Interval) -> Range<u64> {
+    pub fn report_interval(&self, interval: &Interval) -> Range<u64> {
         const REPORT_STORAGE_MAX_FUTURE_TIME_SKEW: u64 = 300;
         // This is a portion of the interval which is guaranteed to be a valid report time
         // provided that the interval start time is valid.
-        interval.start..interval.start + REPORT_STORAGE_MAX_FUTURE_TIME_SKEW
+        let start = std::cmp::max(interval.start, self.task_config.not_before);
+        let end = std::cmp::min(
+            start + REPORT_STORAGE_MAX_FUTURE_TIME_SKEW,
+            self.task_config.not_after,
+        );
+        start..end
     }
```

Comment:
@jhoyla FWIW I think this might have been the bug you were chasing.

Reply:
So I did find a bug here, if you look at #755 you'll see I messed with this. The bug I was hitting is that in the e2e tests taskprov tasks were not being correctly provisioned. If the task isn't found when the leader looks it up in storage then a default task is returned, which has the current time as the `not_before`. This change makes it less likely that you'll hit an issue because the test takes less than a second to run, but doesn't actually fix the underlying.

Reply:
Hmm, what was failing exactly?

If the Client uploads a report for task1, but the Leader resolves task2, then the report should get rejected at some point.

It would be really helpful if you could create a PR, either with the fix, a test case that exercises the bug, or both.